client/src/components/BlockEditor/BlockEditorEditing.tsx

```diff
@@ -201,7 +210,29 @@ export class BlockEditorEditingPresentational extends React.Component<
     }
   };
 
-  private onChangeCallback = (c: any) => {
+  private onChangeCallback = (c: Change) => {
+    // first check to see if document changed, which we can do with !== b/c
+    // Slate uses immutable objects, if it has changed then normalize wrt
+    // pointer spacing
```

Comment:
I appreciate the comments, but one thing that I would have done first is just to write this as: 
```
const documentHasChanged = c.value.document !== this.props.block.value.document
c = documentHasChanged ? normalizeExportSpacing(c) : c
```
or similar